src/main/scala/mesosphere/marathon/api/GroupApiService.scala

```diff
@@ -0,0 +1,71 @@
+package mesosphere.marathon
+package api
+
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.plugin.auth._
+import mesosphere.marathon.raml.{ GroupConversion, Raml }
+import mesosphere.marathon.state._
+
+import scala.async.Async._
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GroupApiService(groupManager: GroupManager)(implicit authorizer: Authorizer, executionContext: ExecutionContext) {
+
+  /**
+    * Encapsulates the group update logic that is following:
+    * - if version is set, group is reverted to that version
+    * - if scaleBy is set, group is scaled up
+    * - if none version nor scaleBy is set then the group is updated
+    */
+  @SuppressWarnings(Array("all")) // async/await
+  def getUpdatedGroup(
```

Comment:
this is a piece of code I did not alter much just took it from the resource class and move it to service class so that I can reuse it between resource and controller.

I also added tests for this logic because it feels to me that this is quite complex business logic but it was not covered by tests at all.

In the future, we can talk about using Either or actually do this in more functional way...

Reply:
The name of this method is a bit confusing. It says get something, and performs side effects in the same time.

Reply:
that would be nice, but it would require bigger refactoring because some code inside those methods actually expect these raml types. Btw. this is not really a domain service, I intentionally placed it to the api package, so that it is very close to API.

Reply:
I'm not sure if we should already introduce service classes. So far we've been copy-pasting existing resource logic into controllers (as bad as it is) with an option of extracting this later.

Reply:
well this code is quite complex and contains a lot of logic that should be tested... and you just cannot test it that easy without extracting it to a separate class...

But I can remove the class and delete the tests if you all think it is not a good idea